prov/psm3/src/psmx3_attr.c

```diff
@@ -286,7 +286,8 @@ static long get_psm3_env(const char *var, int default_value) {
 #endif
 static uint64_t psmx3_check_fi_hmem_cap(void) {
 #ifdef PSM_CUDA
-	if (get_psm3_env("PSM3_CUDA", 0) || get_psm3_env("PSM3_GPUDIRECT", 0))
+	if ((get_psm3_env("PSM3_CUDA", 0) || get_psm3_env("PSM3_GPUDIRECT", 0)) &&
+	    !ofi_hmem_p2p_disabled())
```

Comment:
@acgoldma - please review this proposed change.

Reply:
Sorry, I forgot to hit approve.